app/helpers/assignment_result_helper.rb

```diff
@@ -17,7 +17,7 @@ def assignment_status(assignment)
     elsif assignment.exercise.choices?
       assignment.passed? ? :correct_answer : :wrong_answer
     else
-      assignment.status
```

Comment:
This should work with a submission, not an assignment.

Reply:
In this case it's a bit hard because the submission doesn't know the exercise (because of how composed_of works)